app/models/mixins/authentication_mixin.rb

```diff
@@ -12,6 +12,33 @@ def self.authentication_check_schedule
       assocs = zone.respond_to?(assoc) ? zone.send(assoc) : []
       assocs.each { |a| a.authentication_check_types_queue(:attempt => 1) }
     end
+
+    def self.queue_authentication_check(args, user_id, zone)
```

Comment:
@AparnaKarve I have done manual testing for all of the cloud providers to ensure that it gets the actual result of the validation, and not the result of the queued validation operation. I will take an additional look, however, to ensure that is the case.